extensions/mssql/src/hdfs/ui/hdfsManageAccessDialog.ts

```diff
@@ -0,0 +1,338 @@
+				// = Others permissions section =
+				// ==============================
+
+				const othersPermissionsHeaderRow = modelView.modelBuilder.flexContainer().withLayout({ flexFlow: 'row' }).component();
+				const ownersHeaderCell = modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({ value: allOthersHeader }).component();
+				othersPermissionsHeaderRow.addItem(ownersHeaderCell, { flex: '1 1 auto', CSSStyles: { 'width': `${permissionsNameColumnWidth}px`, 'min-width': `${permissionsNameColumnWidth}px`, ...cssStyles.tableHeaderCss } });
+
+				rootContainer.addItem(othersPermissionsHeaderRow, { CSSStyles: { 'padding-left': '10px', 'box-sizing': 'border-box', 'user-select': 'text' } });
+
+				// Empty initially - this is eventually going to be populated with the "Everyone" permissions
+				this.othersPermissionsContainer = modelView.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
+				rootContainer.addItem(this.othersPermissionsContainer);
+
+				// ===========================
+				// = Add User Or Group Input =
+				// ===========================
+
+				const addUserOrGroupTitle = modelView.modelBuilder.text()
+					.withProperties<azdata.TextComponentProperties>({ value: addUserOrGroupHeader, CSSStyles: { 'margin-block-start': '0px', 'margin-block-end': '10px' } })
+					.component();
+				rootContainer.addItem(addUserOrGroupTitle, { CSSStyles: { 'margin-top': '15px', 'padding-left': '10px', ...cssStyles.titleCss } });
+
+				const typeContainer = modelView.modelBuilder.flexContainer().withProperties({ flexFlow: 'row' }).component();
+				const aclEntryTypeGroup = 'aclEntryType';
+				const userTypeButton = this.createRadioButton(modelView.modelBuilder, userLabel, aclEntryTypeGroup, AclEntryType.user);
+				const groupTypeButton = this.createRadioButton(modelView.modelBuilder, groupLabel, aclEntryTypeGroup, AclEntryType.group);
+				userTypeButton.checked = true;
+				this.addUserOrGroupSelectedType = AclEntryType.user;
+
+				typeContainer.addItems([userTypeButton, groupTypeButton], { flex: '0 0 auto' });
+				rootContainer.addItem(typeContainer, { flex: '0 0 auto' });
+				const addUserOrGroupInputRow = modelView.modelBuilder.flexContainer().withLayout({ flexFlow: 'row' }).component();
+
+				this.addUserOrGroupInput = modelView.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({ inputType: 'text', placeHolder: enterNamePlaceholder, width: 250 }).component();
+				const addUserOrGroupButton = modelView.modelBuilder.button().withProperties<azdata.ButtonProperties>({ label: addLabel, width: 75 }).component();
+				addUserOrGroupButton.onDidClick(() => {
+					this.hdfsModel.createAndAddAclEntry(this.addUserOrGroupInput.value, this.addUserOrGroupSelectedType);
+					this.addUserOrGroupInput.value = '';
+				});
+				addUserOrGroupButton.enabled = false; // Init to disabled since we don't have any name entered in yet
+				this.addUserOrGroupInput.onTextChanged(() => {
```

Comment:
make sure all event handler are disposed properly when the dialog is closed

Reply:
Need to follow up with you on this - it looks like we don't have any sort of onClose event for the dialog so there's currently no way to do this. Will address in follow-up PR since that will likely require more changes than I want to add here.